serve.go

```diff
@@ -102,9 +103,50 @@ func (s *ServeCommand) startHeartbeat() error {
 		if err != nil {
 			s.logger.Errorf("Unable to send initial heartbeat: %v", err)
 		}
+
+		var tmp []string
+		for _, ur := range s.config.IcingaConfig.URL {
+			st := strings.Split(ur, ",")
+			for _, b := range st {
+				tmp = append(tmp, b)
+			}
+		}
+		s.config.IcingaConfig.URL = tmp
+
 		for ts := range s.heartbeatTicker.C {
 			if err := s.heartbeat(ts); err != nil {
 				s.logger.Errorf("sending heartbeat: %s", err)
+
+				// In some rare cases there could be an Icinga2 reload for a config update and the API is not reachable.
+				// So, the switch off of the Config-Master would take in place, but we don't want an unnecessary switch
+				// to the secondary Icinga2 instance.
+				s.logger.Infof("Waiting 5sec for reconnect")
+				time.Sleep(5 * time.Second)
```

Comment:
having a hard-coded 5 second sleep in a loop with an unknown (configurable at runtime) interval seems risky, although 5 seconds is probably low enough that we'll never run into issues with missing heartbeat updates, as running the heartbeat on an interval < 30 seconds probably doesn't make a lot of sense for most setups.